packages/core/internal/ApiDecorator/ApiDecorator.js

```diff
@@ -12,7 +12,9 @@ import hoc from '../../hoc';
 
 // Gets a property from `provider`
 const get = (provider, name) => () => {
-	if (provider) {
+	if (typeof provider[name] === 'function') {
```

Comment:
If we do want to go this route, I would like to see a test that validates this is working.  Perhaps a better check would be to see if the function is bound or not (can check to see if it has an own property of `prototype`, I think).